app/services/schools/register_ect.rb

```diff
@@ -60,7 +61,12 @@ def register!
   private
 
     def already_registered_as_an_ect?
-      ::Teacher.find_by_trn(trn)&.ect_at_school_periods&.exists?
+      # Check if ECT is already registered at THIS school to prevent duplicates
+      # School transfers to different schools are allowed
+      teacher = ::Teacher.find_by_trn(trn)
+      return false unless teacher
```

Comment:
I don't know if it makes sense to do this here, if there's no `teacher`, `#create_teacher!` should `fail`.

Reply:
Yeah, I think it can go. If we get here with no `teacher` we have bigger problems! :smile: